packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -357,23 +365,60 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
+     * at the conclusion of an L2->L1 token transfer via the canonical token bridge.
+     */
+    function sync() public {
+        // Check if the l1Token balance of the contract is greater than the liquidReserves. If it is then the bridging
+        // action from L2->L1 has concluded and the local accounting can be updated.
+        uint256 l1TokenBalance = l1Token.balanceOf(address(this));
+        if (l1TokenBalance > liquidReserves) {
+            // utilizedReserves can go to less than zero. This will happen if the accumulated fees exceeds the current
+            // outstanding utilization. In other words, if outstanding bridging transfers are 0 then utilizedReserves
+            // will equal the total LP fees accumulated over all time.
+            utilizedReserves -= int256(l1TokenBalance - liquidReserves);
+            liquidReserves = l1TokenBalance;
+        }
+    }
 
     /**
      * @notice Computes the exchange rate between LP tokens and L1Tokens. Used when adding/removing liquidity.
      */
-    function exchangeRateCurrent() public view returns (uint256) {
+    function exchangeRateCurrent() public returns (uint256) {
         if (totalSupply() == 0) return 1e18; //initial rate is 1 pre any mint action.
 
-        // Consider a naive rate implementation. This acts like a step function, increasing when funds hit L1 from the
-        // canonical bridge. TODO: update with a more elaborate technique that pays out gradually over the 1 week loan.
+        // First, update fee counters and local accounting of finalized transfers from L2->L1.
+        updateFeeCounters(0); // Accumulate all allocated fees from the last time this method was called.
+        sync(); // Fetch any balance changes due to token bridging finalization and factor them in.
+
+        // ExchangeRate := (liquidReserves+utilizedReserves-undistributedLpFees)/lpTokenSupply
+        // Note to accommodate negative utilizedReserves without using FixedPoint.Signed we need to do a bit of
+        // branching logic. This is a gas optimization so we don't need to import this extra library logic.
+        FixedPoint.Unsigned memory numerator =
+            FixedPoint.Unsigned(liquidReserves).sub(FixedPoint.Unsigned(undistributedLpFees));
+        if (utilizedReserves > 0) numerator = numerator.add(FixedPoint.Unsigned(uint256(utilizedReserves)));
+        else numerator = numerator.sub(FixedPoint.Unsigned(uint256(utilizedReserves * -1)));
+        return numerator.div(FixedPoint.Unsigned(totalSupply())).rawValue;
+    }
+
+    /************************************
+     *           View FUNCTIONS         *
+     ************************************/
+
+    /**
+     * @notice Computes the current amount of unallocated fees that have accumulated from the previous time this the
+     * contract was called.
+     */
+    function getUnallocatedAccumulatedFees() public view returns (uint256) {
```

Comment:
nit: maybe we should change the function name to `getFees` or `getFeePayout` (or something simple like that)? I found the name to be a bit hard to understand.

Reply:
`getFees` sounds good! Have refactored